descwl_shear_sims/simple_sim.py

```diff
@@ -576,12 +587,20 @@ def _setup_stars(
 
         if self.stars:
             assert self.stars_type in ('sample', 'fixed')
-            """
-            if self.stars_type == 'sample':
-                assert self.gals_type == 'wldeblend', (
-                    'gal type must be wldeblend for star type sample',
+            check_keys(
+                self.stars_kws,
+                ('density', 'mag', 'min_mag'),
+                'stars_kws',
+            )
+            if ('density' in self.stars_kws and
+                    isinstance(self.stars_kws['density'], dict)):
+                check_keys(
+                    self.stars_kws['density'],
+                    ('min_density', 'max_density'),
+                    'stars_kws["density"]',
```

Comment:
Can we pull the list of valid keys from the defaults? This will make maintenance easier.

Reply:
Or at least we could store them at the top of the file.

Reply:
they are not in the defaults

Reply:
top of the file would be good too.